loadbalancer/algorithm.go

```diff
@@ -89,19 +95,61 @@ func (*consistentHash) Apply(ctx *routing.LBContext) routing.LBEndpoint {
 	key := net.RemoteHost(ctx.Request).String()
 	if _, err := h.Write([]byte(key)); err != nil {
 		log.Errorf("Failed to write '%s' into hash: %v", key, err)
-		return ctx.Route.LBEndpoints[rand.Intn(len(ctx.Route.LBEndpoints))]
+		return ctx.Route.LBEndpoints.At(rand.Intn(ctx.Route.LBEndpoints.Length()))
 	}
 	sum = h.Sum32()
-	choice := int(sum) % len(ctx.Route.LBEndpoints)
+	choice := int(sum) % ctx.Route.LBEndpoints.Length()
 	if choice < 0 {
-		choice = len(ctx.Route.LBEndpoints) + choice
+		choice = ctx.Route.LBEndpoints.Length() + choice
+	}
+	return ctx.Route.LBEndpoints.At(choice)
+}
+
+// Paper http://www.eecs.harvard.edu/~michaelm/NEWWORK/postscripts/twosurvey.pdf
+type powerOfRandomNChoices struct {
+	rand            *rand.Rand
+	numberOfChoices int
+	mutex           sync.Mutex
+}
+
+func (a *powerOfRandomNChoices) GetScore(e routing.LBEndpoint) int {
+	// TODO: support more metrics. E.g., least connections.
+	// endpoints with higher inflight request should have lower score
+	return -e.Metrics.GetInflightRequests()
```

Comment:
@topicus Is there any specific reason to expose the inflight requests as negative integers? What about 0?

Reply:
@addityasingh see comment above and line 143

Reply:
I understand the comparison is correct but I am specifically interested in understanding why comparing negative integers for inflight request count

Reply:
Because if you read the code in Algorithm the comparison (L143) looks nicer imo.